### PR TITLE
[Hub Generated] Review request for Microsoft.Media to add version preview/2021-05-01-preview

### PR DESCRIPTION
**Pull request by giakas**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/giakas/azure-rest-api-specs/dev-videoanalyzer-Microsoft.Media-2021-05-01-preview...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [x] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by giakas** [comment]
one version of the swagger is already signed off by ARM team in march in private repo, here is the PR with ARM sign off on private repo.
https://github.com/Azure/azure-rest-api-specs-pr/pull/2776

new changes are changes made after the ARM SignOff  
https://github.com/Azure/azure-rest-api-specs/pull/13838/files/e6fd23df57690e50cf219b2fed42db82bb77c6a7..0e164d06487bdb3ef79fff71be88e88f1661e05e

File: specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json [hunk 1]
```diff
@@ -0,0 +1,1033 @@
...
+          }
+        ],
+        "x-ms-examples": {
+          "Update video entity.": {
+            "$ref": "examples/video-patch.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/videoAnalyzers/{accountName}/videos/{videoName}/streamingAuthorization": {
```
Comment (majastrz) [inline specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json:696]: >streamingAuthorization [](start = 146, length = 22)

This doesn't follow the ARM segment naming pattern. It should either be:
`POST /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/videoAnalyzers/{accountName}/videos/{videoName}/getStreamingAuthorization`
OR
`GET /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/videoAnalyzers/{accountName}/videos/{videoName}/streamingAuthorizations/default` (instread of `default` you can also use `latest`, `current`, `main`)

**Comment by majastrz** [comment]
Added one comment. Otherwise, the changes look good to me.

File: specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json (hunk 1 above)
Comment (giakas) [reply specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json]: @majastrz  we have changed these to post calls with List word in the name

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

File: specification/videoanalyzer/resource-manager/readme.csharp.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Please also specify `--csharp-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-net clone>`.
+
+```yaml $(csharp)
+csharp:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  payload-flattening-threshold: 1
+  clear-output-folder: true
+  client-side-validation: false
+  namespace: Microsoft.Media
```
Comment (allenjzhang) [inline specification/videoanalyzer/resource-manager/readme.csharp.md:13]: The namespace doesn't look right for mgmt plane SDK.

File: specification/videoanalyzer/resource-manager/readme.csharp.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  output-folder: $(csharp-sdks-folder)/videoanalyzer/management/Microsoft.Media/GeneratedProtocol
```
Comment (allenjzhang) [inline specification/videoanalyzer/resource-manager/readme.csharp.md:14]: The path doesn't look right either.

File: specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json [hunk 4]
```diff
@@ -0,0 +1,1035 @@
...
+        "x-ms-examples": {
+          "Update video entity.": {
+            "$ref": "examples/video-patch.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/videoAnalyzers/{accountName}/videos/{videoName}/listStreamingToken": {
+      "post": {
+        "summary": "Generates a streaming token for video playback.",
```
Comment (allenjzhang) [inline specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json:700]: >Generates [](start = 20, length = 9)

nit: Generate? or List

File: specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json [hunk 5]
```diff
@@ -0,0 +1,1035 @@
...
+  },
+  "parameters": {
+    "SubscriptionId": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The unique identifier for a Microsoft Azure subscription."
+    },
+    "ResourceGroupName": {
```
Comment (allenjzhang) [inline specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json:1011]: nit: use the common-type's parameter?

File: specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/VideoAnalyzers.json [hunk 6]
```diff
@@ -0,0 +1,1048 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "Impersonate your user account"
+      }
+    }
+  },
+  "definitions": {
+    "Provider": {
```
Comment (allenjzhang) [inline specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/VideoAnalyzers.json:29]: >Provider [](start = 5, length = 8)

Where is this referenced?

File: specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json (hunk 4 above)
Comment (giakas) [reply specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/Videos.json:700]: even though the API says list, its a post operation, which generates new token every time.

File: specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/VideoAnalyzers.json (hunk 6 above)
Comment (giakas) [reply specification/videoanalyzer/resource-manager/Microsoft.Media/preview/2021-05-01-preview/VideoAnalyzers.json:29]: removed.

File: specification/videoanalyzer/resource-manager/readme.csharp.md (hunk 2 above)
Comment (giakas) [reply specification/videoanalyzer/resource-manager/readme.csharp.md:13]: fixed

Comment (giakas) [reply specification/videoanalyzer/resource-manager/readme.csharp.md:13]: fixed

File: specification/videoanalyzer/resource-manager/readme.csharp.md (hunk 3 above)
Comment (giakas) [reply specification/videoanalyzer/resource-manager/readme.csharp.md:14]: fixed

**Review by ArcturusZhang — APPROVED** [review]
Approved for the changes in `readme.go.md`

**Review by allenjzhang — APPROVED** [review]

**Pull request merged**